GVRf/Framework/jni/objects/components/camera_rig_jni.cpp

```diff
@@ -126,49 +120,24 @@ Java_org_gearvrf_NativeCameraRig_getLookAt(JNIEnv * env,
 JNIEXPORT jlong JNICALL
 Java_org_gearvrf_NativeCameraRig_ctor(JNIEnv * env,
         jobject obj) {
-    std::shared_ptr<CameraRig> camera_rig(new CameraRig());
-    return reinterpret_cast<jlong>(new std::shared_ptr<CameraRig>(camera_rig));
+    return reinterpret_cast<jlong>(new CameraRig());
 }
 
 JNIEXPORT jint JNICALL
 Java_org_gearvrf_NativeCameraRig_getCameraRigType(JNIEnv * env,
         jobject obj, jlong jcamera_rig) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
+    CameraRig* camera_rig = reinterpret_cast<CameraRig*>(jcamera_rig);
     return static_cast<jint>(camera_rig->camera_rig_type());
 }
 
 JNIEXPORT void JNICALL
 Java_org_gearvrf_NativeCameraRig_setCameraRigType(JNIEnv * env,
         jobject obj, jlong jcamera_rig, jint camera_rig_type) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
+    CameraRig* camera_rig = reinterpret_cast<CameraRig*>(jcamera_rig);
     camera_rig->set_camera_rig_type(
             static_cast<CameraRig::CameraRigType>(camera_rig_type));
 }
 
-JNIEXPORT jlong JNICALL
-Java_org_gearvrf_NativeCameraRig_getLeftCamera(JNIEnv * env,
-        jobject obj, jlong jcamera_rig) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
-    std::shared_ptr<Camera> left_camera(camera_rig->left_camera());
-    return left_camera.get() == NULL ?
-            0 :
-            reinterpret_cast<jlong>(new std::shared_ptr<Camera>(left_camera));
-}
-
-JNIEXPORT jlong JNICALL
-Java_org_gearvrf_NativeCameraRig_getRightCamera(JNIEnv * env,
-        jobject obj, jlong jcamera_rig) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
-    std::shared_ptr<Camera> right_camera(camera_rig->right_camera());
-    return right_camera.get() == NULL ?
-            0 :
-            reinterpret_cast<jlong>(new std::shared_ptr<Camera>(right_camera));
-}
-
```

Comment:
Read The Source, Luke.

From: chenchao1407 [mailto:notifications@github.com]
Sent: Tuesday, May 12, 2015 1:44 PM
To: Samsung/GearVRf
Cc: Jon Shemitz
Subject: Re: [GearVRf] GVRF 2.0! (#41)


In GVRf/Framework/jni/objects/components/camera_rig_jni.cpp<https://github.com/Samsung/GearVRf/pull/41#discussion_r30178164>:

> -    return left_camera.get() == NULL ?

> -            0 :

> -            reinterpret_cast<jlong>(new std::shared_ptr<Camera>(left_camera));

> -}

> -

> -JNIEXPORT jlong JNICALL

> -Java_org_gearvrf_NativeCameraRig_getRightCamera(JNIEnv * env,

> -        jobject obj, jlong jcamera_rig) {

> -    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<

> -            CameraRig>*>(jcamera_rig);

> -    std::shared_ptr<Camera> right_camera(camera_rig->right_camera());

> -    return right_camera.get() == NULL ?

> -            0 :

> -            reinterpret_cast<jlong>(new std::shared_ptr<Camera>(right_camera));

> -}

> -

Why do we need to remove these two methods since we still keep right/left_camera() method to get the pointer of two cameras?

—
Reply to this email directly or view it on GitHub<https://github.com/Samsung/GearVRf/pull/41/files#r30178164>.